scss/_badge.scss

```diff
@@ -1,42 +1,106 @@
-// Base class
 //
-// Requires one of the contextual, color modifier classes for `color` and
-// `background-color`.
+// Base styles
+//
 
 .badge {
   // scss-docs-start badge-css-vars
-  --#{$prefix}badge-padding: #{subtract(var(--#{$prefix}badge-padding-y), 1px) var(--#{$prefix}badge-padding-x) add(var(--#{$prefix}badge-padding-y), 1px)}; // OUDS mod
-  --#{$prefix}badge-padding-x: #{$badge-padding-x};
-  --#{$prefix}badge-padding-y: #{$badge-padding-y};
-  @include rfs($badge-font-size, --#{$prefix}badge-font-size);
-  --#{$prefix}badge-font-weight: #{$badge-font-weight};
-  --#{$prefix}badge-color: #{$badge-color};
-  --#{$prefix}badge-border-radius: #{$badge-border-radius};
+  --#{$prefix}badge-size-width: #{$ouds-badge-size-min-width-medium};
+  --#{$prefix}badge-size-min-width: #{$ouds-badge-size-min-width-medium};
+  --#{$prefix}badge-size-max-width: #{$ouds-badge-size-max-width-medium-dot};
+  --#{$prefix}badge-size-height: #{$ouds-badge-size-max-height-medium};
+  --#{$prefix}badge-size-min-height: #{$ouds-badge-size-max-height-medium};
+  --#{$prefix}badge-size-max-height: #{$ouds-badge-size-max-height-medium};
+  --#{$prefix}badge-color: #{$ouds-color-content-on-status-emphasized-alt};
+  --#{$prefix}badge-background-color: #{$ouds-color-surface-status-neutral-emphasized};
+  --#{$prefix}badge-border-radius: #{$ouds-badge-border-radius-pill};
+  --#{$prefix}badge-icon-size-width: 10px;
+  --#{$prefix}badge-icon-size-height: 10px;
+  --#{$prefix}badge-count-padding-x: 0;
   // scss-docs-end badge-css-vars
 
-  display: inline-block;
-  padding: var(--#{$prefix}badge-padding);
-  @include font-size(var(--#{$prefix}badge-font-size));
-  font-weight: var(--#{$prefix}badge-font-weight);
-  line-height: 1;
+  display: inline-flex;
```

Comment:
Do we want it flex with alignments or block with padding ? (We have padding tokens)